### PR TITLE
Fix issues with EqualValueBoundaryConstraint

### LINKED ISSUE #7015: Problems with EqualValueBoundaryConstraint
### Description of the enhancement or error report
There are two issues with EqualValueBoundaryConstraint:
- If a boundary is provided to define the slave nodes, the set of slave nodes doesn't get updated when the mesh changes
- The automatic selection of the master node from the slave boundary nodes was picking the first element in the vector, which wasn't always the same on all processors in parallel.

### Rationale for the enhancement or information for reproducing the error
The first issue will permit using this with XFEM and adaptivity.  The second issue was giving buggy results in parallel.

### Identified impact
Changes internal to EqualValueBoundaryConstraint.C

### PR DESCRIPTION
**Pull request by bwspenc**
closes #7015

### CODE REVIEW

File: framework/src/constraints/EqualValueBoundaryConstraint.C [hunk 1]
```diff
@@ -37,50 +37,68 @@ EqualValueBoundaryConstraint::EqualValueBoundaryConstraint(const InputParameters
...
     std::vector<dof_id_type>::iterator in;
 
-    //Setting master node to first node of the slave node set if no master node id is pprovided
+    //Set master node to first node of the slave node set if no master node id is provided
+    //_master_node_vector defines master nodes in the base class
     if (_master_node_id == std::numeric_limits<unsigned int>::max())
     {
-      in = nodelist.begin();
-      _master_node_id = *in;
+      in = std::min_element(nodelist.begin(),nodelist.end());
```
Comment (andrsd) [inline framework/src/constraints/EqualValueBoundaryConstraint.C]: Space after `,`

**Comment by andrsd** [comment]
Adaptivity was not considered when `NodalConstraint` system was added, neither parallel execution. Nice to see that fixed so easily...

**Comment by jwpeterson** [comment]
MOOSE/task labels are for issues, right?

**Comment by andrsd** [comment]
> MOOSE/task labels are for issues, right?

Yes, a case of brain damage ;-)

**Pull request merged**